hopsworks-ear/test/spec/execution_spec.rb

```diff
@@ -18,121 +18,127 @@
+                json_body[:state].eql? "KILLED"
+              end
             end
+            it "should fail to start two executions in parallel" do
+              $job_name_3 = "demo_job_3_" + type
+              create_sparktour_job(@project, $job_name_3, type)
+              start_execution(@project[:id], $job_name_3)
+              start_execution(@project[:id], $job_name_3)
+              expect_status(400)
+              expect_json(errorCode: 130010)
+              stop_execution(@project[:id], $job_name_3)
+            end
+            it "should run job and get out and err logs" do
+              $job_name_4 = "demo_job_4_" + type
+              create_sparktour_job(@project, $job_name_4, type)
+              start_execution(@project[:id], $job_name_4)
+              execution_id = json_body[:id]
+              expect_status(201)
 
-            stop_execution(@project[:id], $job_name)
-            expect_status(200)
-            wait_for_execution do
-              get_execution(@project[:id], $job_name, execution_id)
-              json_body[:state].eql? "KILLED"
-            end
-          end
-          it "should fail to start two executions in parallel" do
-            create_sparktour_job(@project, $job_name)
-            start_execution(@project[:id], $job_name)
-            start_execution(@project[:id], $job_name)
-            expect_status(400)
-            expect_json(errorCode: 130010)
-          end
-          it "should run job and get out and err logs" do
-            create_sparktour_job(@project, $job_name)
-            start_execution(@project[:id], $job_name)
-            execution_id = json_body[:id]
-            expect_status(201)
+              wait_for_execution do
+                get_execution(@project[:id], $job_name_4, execution_id)
+                json_body[:state].eql? "FINISHED"
+              end
 
-            wait_for_execution do
-              get_execution(@project[:id], $job_name, execution_id)
-              json_body[:state].eql? "FINISHED"
-            end
+              #wait for log aggregation
+              wait_for_execution do
+                get_execution_log(@project[:id], $job_name_4, execution_id, "out")
+                json_body[:log] != "No log available"
+              end
 
-            #wait for log aggregation
-            wait_for_execution do
-              get_execution_log(@project[:id], $job_name, execution_id, "out")
-              json_body[:log] != "No log available"
-            end
+              #get out log
+              get_execution_log(@project[:id], $job_name_4, execution_id, "out")
+              expect(json_body[:type]).to eq "OUT"
+              expect(json_body[:log]).to be_present
 
-            #get out log
-            get_execution_log(@project[:id], $job_name, execution_id, "out")
-            expect(json_body[:type]).to eq "OUT"
-            expect(json_body[:log]).to be_present
+              #wait for log aggregation
```

Comment:
@robzor92 why `wait for log aggregation` is now after `get out log`?

Reply:
There are two wait for log aggregations, one for err and one for _out_.

Reply:
Aha ok then